lib/blacklight/access_controls/enforcement.rb

```diff
@@ -25,72 +29,61 @@ module Enforcement
-        user_access_filters = []
-
-        # Grant access based on user id & group
-        solr_access_filters_logic.each do |method_name|
-          user_access_filters += send(method_name, permission_types, ability)
-        end
-        user_access_filters
-      end
-
-      #
-      # Solr query modifications
-      #
-
-      # Set solr_parameters to enforce appropriate permissions
-      # * Applies a lucene query to the solr :q parameter for gated discovery
-      # * Uses public_qt search handler if user does not have "read" permissions
-      # @param solr_parameters the current solr parameters
-      def add_access_controls_to_solr_params(solr_parameters)
-        apply_gated_discovery(solr_parameters)
-      end
-
       # Which permission levels (logical OR) will grant you the ability to discover documents in a search.
-      # Override this method if you want it to be something other than the default
+      # Override this method if you want it to be something other than the default, or hit the setter
       def discovery_permissions
         @discovery_permissions ||= %w(discover read)
       end
 
-      def discovery_permissions=(permissions)
-        @discovery_permissions = permissions
+      protected
+
+      # Grant access based on user id & group
+      # @return [Array{Array{String}}]
+      def gated_discovery_filters(permission_types = discovery_permissions, ability = current_ability)
+        solr_access_filters_logic.map { |method| send(method, permission_types, ability).reject(&:blank?) }
       end
 
-      # Controller before filter that sets up access-controlled lucene query in order to provide gated discovery behavior
-      # @param solr_parameters the current solr parameters
+      ### Solr query modifications
+
+      # Controller before_filter that sets up access-controlled lucene query to provide gated discovery behavior.
+      # Set solr_parameters to enforce appropriate permissions.
+      # @param [Hash{Object}] solr_parameters the current solr parameters, to be modified herein!
+      # @note Applies a lucene filter query to the solr :fq parameter for gated discovery.
       def apply_gated_discovery(solr_parameters)
         solr_parameters[:fq] ||= []
-        solr_parameters[:fq] << gated_discovery_filters.join(' OR ')
+        solr_parameters[:fq] << gated_discovery_filters.reject(&:blank?).join(' OR ')
         Rails.logger.debug("Solr parameters: #{solr_parameters.inspect}")
       end
 
+      # For groups
+      # @return [Array{String}] values are lucence syntax term queries suitable for :fq
+      # @example
+      #   [ "({!terms f=discover_access_group_ssim}public,faculty,africana-faculty,registered)",
+      #     "({!terms f=read_access_group_ssim}public,faculty,africana-faculty,registered)" ]
       def apply_group_permissions(permission_types, ability = current_ability)
-        # for groups
+        groups = ability.user_groups
+        return [] if groups.empty?
```

Comment:
Seeing as how Ability has default groups and even anonymous public users have a group, I might also accept throwing an error here.  But if you do intentionally pass "no group filters", what do you expect?  Multiple `:fq`'s against nothing or no group filters?  

If you don't have groups then you would be creating filter clauses of, e.g. `({!terms f=my_field_ssim})`.  That matches nothing in a lucene query, because lucene doesn't index nothing.  But more broadly, the end result is an OR query.  After a single populated query, including additional no-match clauses is worthless.  Without groups, the compound `fq` is going to say, e.g.: 
`'discover_access_person_ssim:user_1@abc.com' OR 'read_access_person_ssim:user_1@abc.com'`

So that cannot match *more* than adding a bunch of other OR clauses.

Concrete queries:
[{!terms%20f=format_main_ssim}](https://sul-solr-a.stanford.edu/solr/exhibits_stage/select?fq={!terms%20f=format_main_ssim}&indent=on&wt=json) -- field exists, query nothing, get nothing
[{!terms%20f=format_main_ssim}Book](https://sul-solr-a.stanford.edu/solr/exhibits_stage/select?fq={!terms%20f=format_main_ssim}Book&indent=on&wt=json) -- field exists, query value, gets results
[{!terms f=no_such_field}](https://sul-solr-a.stanford.edu:443/solr/exhibits_stage/select?fq={!terms f=no_such_field}&indent=on&wt=json) -- field non-existent, query nothing, get nothing